addon/components/nypr-card/button.js

```diff
@@ -0,0 +1,15 @@
+import Ember from 'ember';
+import layout from '../../templates/components/nypr-card/button';
+
+const NyprCardButton = Ember.Component.extend({
+  layout,
+  tagName: 'button',
+  classNames: ['nypr-card-button'],
+  'data-test-selector': 'nypr-card-button'
+});
+
+NyprCardButton.reopen({
+  attributeBindings: ['data-test-selector']
+});
```

Comment:
They use reopen in that example because they're extending the Ember builtin `link-to`  and `input` components. For your own components, just adding `attributeBindings` to the component properties is fine. https://emberjs.com/api/classes/Ember.Component.html#toc_html-attributes